babel/messages/extract.py

```diff
@@ -520,103 +512,147 @@ def extract_python(
     # currently parsing one.
     current_fstring_start = None
 
-    for tok, value, (lineno, _), _, _ in tokens:
-        if call_stack == -1 and tok == NAME and value in ('def', 'class'):
+    # Keep the stack of all function calls and its related contextual variables,
+    # so we can handle nested gettext calls.
+    function_stack = []
+    # Keep the last encountered function/variable name for when we encounter
+    # an opening parenthesis
+    last_name = None
+    # Keep track of whether we're in a class or function definition
+    in_def = False
+    # Keep track of whether we're in a block of translator comments
+    in_translator_comments = False
+    # Keep track of the last encountered translator comments
+    translator_comments = []
+    # Keep track of the (split) strings encountered
+    message_buffer = []
+
+    for token, value, (line_no, _), _, _ in tokens:
```

Comment:
Tiny thing, but could the local `line_no` be renamed back to `lineno`? It would make reviewing easier since the diff is smaller 😅 
(Similarly, `line_no` elsewhere should maybe be `lineno` for consistency and compat.)